crawl-ref/source/skills.cc

```diff
@@ -1169,34 +1170,35 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
         if (reduced && you_skill_cost_level)
             ++max_xp;
 
-        int cost = calc_skill_cost(you_skill_cost_level);
+        const int cost = calc_skill_cost(you_skill_cost_level);
         // Maximum number of skill points to transfer in one go.
         // It's max_xp*10/cost rounded up.
-        int max_skp = (max_xp * 10 + cost - 1) / cost;
-        max_skp = max(max_skp, 1);
-        int delta_skp = min<int>(abs((int)(target - you_skill)),
+        const int max_skp = max((max_xp * 10 + cost - 1) / cost, 1);
+        skill_diff delta;
```

Comment:
I don't feel strongly about this, but what about using the `skill_diff(int, int)` constructor you defined? You're initializing to zero and then unconditionally overwriting, which seems a bit weird.

On the other hand I can see how this style might be more readable than what I suggest. Use your judgment on that, I just wanted to make sure you considered both ways.